File: src/java/test/java/simplifying/method/calls/renameMethod.java
package simplifying.method.calls;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class renameMethod {
    @Test
    void john_doe_is_a_model_employee() {
        Employee johnDoe = new Employee("John Doe", "?", "Beach", List.of("Hunter", "Accountant"));

        assertThat(johnDoe.getId()).isNotNull();
        assertThat(johnDoe.getName()).isEqualTo("John Doe");
        assertThat(johnDoe.getRole()).isEqualTo("?");
        assertThat(johnDoe.getCurrentProject()).isEqualTo("Beach");
        assertThat(johnDoe.isProfessionalService()).isTrue();
        assertThat(johnDoe.isIdeal()).isTrue();
        assertThat(johnDoe.hasSomething("Accountant")).isTrue();
        assertThat(johnDoe.hasSomething("Football")).isFalse();
    }
}

File: src/scala/src/test/scala/simplifying/method/calls/renameMethod.scala
package simplifying.method.calls

import org.scalatest.funsuite.AnyFunSuite

class renameMethod extends AnyFunSuite {
  test("john doe is a perfect employee") {
    val johnDoe =
      Employee("John Doe", "?", "Beach", List("Hunter", "Accountant"))

    assert(johnDoe.getId.nonEmpty)
    assert(johnDoe.name == "John Doe")
    assert(johnDoe.role == "?")
    assert(johnDoe.currentProject == "Beach")
    assert(johnDoe.isWorking)
    assert(johnDoe.hasIdealJob)
    assert(johnDoe.isSkilledAt("Accountant"))
    assert(!johnDoe.isSkilledAt("Football"))
  }
}
